PurchasesTests/Caching/InMemoryCachedObjectTests.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Created by RevenueCat on 2/4/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+import Nimble
+import XCTest
+
+@testable import Purchases
+
+class InMemoryCachedObjectTests: XCTestCase {
+
+    // mark: isCacheStale
+
+    func testIsCacheStaleIsFalseBeforeDurationExpires() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.stubbedNow = Date()
+
+        cachedObject.cacheInstance("myString", date: Date())
+
+        expect(cachedObject.isCacheStale()) == false
+
+        cachedObject.stubbedNow = Date(timeIntervalSinceNow: 4)
+
+        expect(cachedObject.isCacheStale()) == false
+    }
+
+    func testIsCacheStaleIsTrueAfterDurationExpires() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.stubbedNow = Date()
+
+        cachedObject.cacheInstance("myString", date: Date())
+
+        expect(cachedObject.isCacheStale()) == false
+
+        cachedObject.stubbedNow = Date(timeIntervalSinceNow: 6)
+
+        expect(cachedObject.isCacheStale()) == true
+    }
+
+    // mark: clearCacheTimestamp
+
+    func testClearCacheTimestampClearsCorrectly() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
+        cachedObject.cacheInstance("myString", date: Date())
+        expect(cachedObject.lastUpdatedAt).toNot(beNil())
+        expect(cachedObject.cachedInstance()).toNot(beNil())
+
+        cachedObject.clearCacheTimestamp()
+        expect(cachedObject.lastUpdatedAt).to(beNil())
+        expect(cachedObject.cachedInstance()).toNot(beNil())
+    }
+
+    // mark: clearCache
+
+    func testClearCacheClearsCorrectly() {
+        let cacheDurationInSeconds: Int32 = 5
+
+        let cachedObject = RCInMemoryCachedObject<NSString>(cacheDurationInSeconds: cacheDurationInSeconds,
+                                                            lastUpdatedAt: nil)
```

Comment:
`lastUpdatedAt` in the `RCInMemoryCachedObject` constructor is always set to`nil`, why don't you just use the constructor that only accepts the cache duration and remove the one that accepts the `lastUpdatedAt`? I believe the `lastUpdatedAt` should be updated through the `cacheInstance` method.

Reply:
yeah, good call. 
I'm getting used to the best ways to inject these things, and I agree that injecting wasn't necessary here. removed.